functions/index.js

```diff
@@ -200,4 +204,102 @@ exports.getArcGISToken = onCall({
     logger.error('Error in getArcGISTokenR:', error);
     throw new HttpsError('internal', 'Internal server error');
   }
+});
+
+//HTTP to get ArcGIS token
+// This function is designed to be called from other backend services or functions
+// It is not intended for direct invocation from the client-side
+// It uses CORS to allow cross-origin requests, which is useful for server-to-server communication
+// The function retrieves credentials from Google Cloud Secret Manager
+exports.getArcGISTokenHttp = functions.https.onRequest(async (req, res) => {
+  // Handle CORS preflight requests and set CORS headers for actual requests.
+  cors(req, res, async () => {
+
+      logger.info("getArcGISTokenHttp function triggered.");
+
+      // IMPORTANT: Authentication/Authorization for server-to-server calls
+      // This function relies on IAM for secure invocation when called from another backend/function.
+      // In Google Cloud Console -> Cloud Functions -> select this function -> Permissions:
+      // Grant the 'Cloud Functions Invoker' role to the service account
+      // of the calling function/service (e.g., the proxy function in borehole-prod).
+      // Cloud Functions infrastructure will handle verifying the OIDC token.
+
+      try {
+          // Get credentials from Secret Manager using the existing helper
+          logger.info("Accessing secrets for ArcGIS token generation...");
+          // Using accessSecret function defined earlier in your file
+          const username = await accessSecret('geolmap_user');
+          const password = await accessSecret('geolmap_pass');
+          logger.info("Secrets accessed successfully.");
+
+          // Prepare request to ArcGIS token endpoint
+          const tokenUrl = 'https://webmaps.geology.utah.gov/arcgis/tokens/generateToken';
+          const params = new URLSearchParams({
+              username: username,
+              password: password,
+              client: 'referer',
+              // Referer might need adjustment depending on how ArcGIS server is configured,
+              // especially if the call originates server-side now.
+              referer: 'https://borehole.geology.utah.gov', // Or your primary app domain
+              expiration: 60, // Token expiration in minutes
+              f: 'json'
+          });
+
+          logger.info("Requesting token from ArcGIS...");
```

Comment:
is this a useful log? can we modify or remove it? where do these logs get viewed?